core/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -60,7 +72,28 @@ protected WritePipelineResponse newResponse() {
 
     @Override
     protected void masterOperation(PutPipelineRequest request, ClusterState state, ActionListener<WritePipelineResponse> listener) throws Exception {
-        pipelineStore.put(clusterService, request, listener);
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+        nodesInfoRequest.clear();
+        nodesInfoRequest.ingest(true);
```

Comment:
BTW I am adding the forgotten filtering to DiscoveryNodes as part of #16963 anyways.